src/compas_viewer/scene/pointobject.py

```diff
@@ -8,12 +8,21 @@
 class PointObject(ViewerSceneObject, GeometryObject):
     """Viewer scene object for displaying COMPAS Point geometry.
 
+    Parameters
+    ----------
+    point : :class:`compas.geometry.Point`
+        The point geometry to display.
+    show_points : bool, optional
+        Whether to display the point in the viewer. Default is True.
+
     See Also
     --------
     :class:`compas.geometry.Point`
     """
 
     def __init__(self, point: Point, **kwargs):
```

Comment:
def __init__(self, point: Point, show_points=True, **kwargs):
    self.show_points=show_points

Reply:
Hi Li. This looks like a more readable solution but I doubt if it works.

if we do:
```python
    def __init__(self, point: Point, show_points=True, **kwargs):
        self.show_points = show_points
        super(PointObject, self).__init__(geometry=point, **kwargs)
        self.geometry: Point
```
then call:
````python
viewer.add(point)
````
it is still not displayed. `kwargs` from the config overwrites the `show_points`..

I tried some ways around and may suggest keep the current:)

Reply:
in this case shouldn't be: 
```
def __init__(self, point: Point, show_points=True, **kwargs):
        super(PointObject, self).__init__(geometry=point, show_points=show_points, **kwargs)
        self.geometry: Point
```

likely something inside `super().__init__` is overwriting your fist line.

Reply:
> in this case shouldn't be:
> 
> ```
> def __init__(self, point: Point, show_points=True, **kwargs):
>         super(PointObject, self).__init__(geometry=point, show_points=show_points, **kwargs)
>         self.geometry: Point
> ```
> 
> likely something inside `super().__init__` is overwriting your fist line.

also doubt if this works... if we rund:

```python
    viewer.add(point, pointscolor=Color(random(), random(), random()), pointssize=30)
```

The points are hidden, meaning only the configuration file is working...

Reply:
please also use `super()` instead of the old version. the old version is only needed for py2 compatibility, which we don't need here...

Reply:
if `self.show_points` is defined on the base scene object, this will never work
```python
def __init__(self, point: Point, show_points: bool = True, **kwargs):
    self.show_points = show_points
    super().__init__(geometry=point, **kwargs)
```
because you overwrite `self.show_points = show_points` again with `super().__init__(geometry=point, **kwargs)`

you should be able to do
```python
def __init__(self, point: Point, show_points: bool = True, **kwargs):
    super().__init__(geometry=point, **kwargs)
    self.show_points = show_points
```
and if that doesn't work, then there is something wrong with the code

but if `self.show_points` is indeed already defined on the base object that should not even be necessary
```python
def __init__(self, point: Point, **kwargs):
    super().__init__(geometry=point, **kwargs)
```
should suffice.
again, if that is not the case, there is something wrong with the code.

Reply:
please add ALL type annotations.
and use `super()` without arguments, as it is intended in py3